### PR TITLE
Og 121 remove 0x

### PR DESCRIPTION
**Pull request by drortirosh**
- use minimal LibBytes of readBytes32/readBytes4
- use "truncate" in-place (so we don't need complex "slice")
- GsnUtils.revertWithData: to revert with raw bytes data (taken from internal call revert()), instead of reason string.
- API: don't extract on-chain revert reason: return entire revert byte array (in TrasnactionRelayed and relayCall return value)
- added decodeRevertReason() in js

### CODE REVIEW

File: contracts/RelayHub.sol [hunk 1]
```diff
@@ -302,7 +300,8 @@ contract RelayHub is IRelayHub {
             bytes memory retData;
             (success, retData) = relayRequest.relayData.paymaster.call{gas:gasLimits.preRelayedCallGasLimit}(vars.data);
             if (!success) {
-                revertWithStatus(RelayCallStatus.RejectedByPreRelayed, GsnEip712Library.getTruncatedData(retData));
+                GsnEip712Library.truncatedData(retData);
```
Comment (shahafn) [inline contracts/RelayHub.sol:303]: Why is it here then?

File: contracts/utils/MinLibBytes.sol [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+// a minimal subset from 0x/LibBytes
+/* solhint-disable no-inline-assembly */
+pragma solidity ^0.6.2;
+
+library MinLibBytes {
+
+    //truncate the given parameter if its length is above the given maximum length
+    // do nothing otherwise.
+    //NOTE: the truncation is done in-place
+    function truncatedData(bytes memory data, uint256 maxlen) internal pure {
```
Comment (shahafn) [inline contracts/utils/MinLibBytes.sol:12]: truncateInPlace maybe?

File: contracts/RelayHub.sol (hunk 1 above)
Comment (drortirosh) [reply contracts/RelayHub.sol]: yes, renamed.

File: contracts/utils/MinLibBytes.sol (hunk 2 above)
Comment (drortirosh) [reply contracts/utils/MinLibBytes.sol]: yes, renamed.

**Pull request merged**